api/dbadapters/snowflake.ts

```diff
@@ -35,35 +35,10 @@ interface ISnowflakeResultStream {
 const snowflake: ISnowflake = require("snowflake-sdk");
 
 export class SnowflakeDbAdapter implements IDbAdapter {
-  private connection: ISnowflakeConnection;
-  private connected: Promise<void>;
+  private connection: Promise<ISnowflakeConnection>;
 
   constructor(credentials: Credentials) {
-    const snowflakeCredentials = credentials as dataform.ISnowflake;
-    this.connection = snowflake.createConnection({
-      account: snowflakeCredentials.accountId,
-      username: snowflakeCredentials.username,
-      password: snowflakeCredentials.password,
-      database: snowflakeCredentials.databaseName,
-      warehouse: snowflakeCredentials.warehouse,
-      role: snowflakeCredentials.role
-    });
-    // We are forced to try our own HTTPS connection to the final <accountId>.snowflakecomputing.com URL
-    // in order to verify its certificate. If we don't do this, and pass an invalid account ID (which thus
-    // resolves to an invalid URL) to the snowflake connect() API, snowflake-sdk will not handle the
-    // resulting error correctly (and thus crash this process).
-    this.connected = this.verifyCertificate(snowflakeCredentials.accountId).then(
-      () =>
-        new Promise<void>((resolve, reject) => {
-          this.connection.connect((err, conn) => {
-            if (err) {
-              reject(err);
-            } else {
-              resolve();
-            }
-          });
-        })
-    );
+    this.connection = connect(credentials as dataform.ISnowflake);
```

Comment:
connectionPromise ?
it seems weird that we await for this with a variable named to `connection`.